services/director-v2/src/simcore_service_director_v2/models/domains/comp_pipelines.py

```diff
@@ -0,0 +1,18 @@
+from typing import Dict, List
+
+from models_library.projects import ProjectID, RunningState
+from pydantic import BaseModel, validator
+
+
+class CompPipelineAtDB(BaseModel):
+    project_id: ProjectID
+    dag_adjacency_list: Dict[str, List[str]]  # json serialization issue if using NodeID
+    state: RunningState
+
+    @validator("dag_adjacency_list", pre=True)
+    @classmethod
+    def auto_convert_dag(cls, v):
```

Comment:
can you add a comment here with an example of why are you enforcing this? I thought this conversion is automatic